client-runtime/src/main/java/com/microsoft/rest/v2/RestProxy.java

```diff
@@ -19,209 +20,169 @@
-                Annotation[] annotations = method.getParameterAnnotations()[i];
-                for (Annotation annotation : annotations) {
-                    if (annotation.annotationType().equals(HostParam.class)) {
-                        String name = ((HostParam) annotation).value();
-                        // TODO: mark encoded using a better approach than positive/negative int
-                        info.hostArgs.put(name, ((HostParam) annotation).encoded() ? i : -i);
-                    }
-                    if (annotation.annotationType().equals(PathParam.class)) {
-                        String name = ((PathParam) annotation).value();
-                        info.pathArgs.put(name, ((PathParam) annotation).encoded() ? i : -i);
-                    }
-                    if (annotation.annotationType().equals(QueryParam.class)) {
-                        String name = ((QueryParam) annotation).value();
-                        info.queryArgs.put(name, ((PathParam) annotation).encoded() ? i : -i);
-                    }
-                    if (annotation.annotationType().equals(HeaderParam.class)) {
-                        String name = ((HeaderParam) annotation).value();
-                        info.headerArgs.put(name, i);
-                    }
-                    if (annotation.annotationType().equals(BodyParam.class)) {
-                        info.bodyArg = i;
-                    }
-                }
-            }
+    /**
+     * Create a proxy implementation of the provided Swagger interface.
+     * @param swaggerInterface The Swagger interface to provide a proxy implementation for.
+     * @param httpClient The internal HTTP client that will be used to make REST calls.
+     * @param serializer The serializer that will be used to convert POJOs to and from request and
+     *                   response bodies.
+     * @param <A> The type of the Swagger interface.
+     * @return A proxy implementation of the provided Swagger interface.
+     */
+    @SuppressWarnings("unchecked")
+    public static <A> A create(Class<A> swaggerInterface, HttpClient httpClient, SerializerAdapter<?> serializer) {
+        final SwaggerInterfaceProxyDetails interfaceProxyDetails = new SwaggerInterfaceProxyDetails();
+
+        String host = null;
+        final Host hostAnnotation = swaggerInterface.getAnnotation(Host.class);
+        if (hostAnnotation != null) {
+            host = hostAnnotation.value();
+        }
+
+        for (Method method : swaggerInterface.getDeclaredMethods()) {
+            final SwaggerMethodProxyDetails methodProxyDetails = interfaceProxyDetails.getMethodProxyDetails(method.getName());
 
             if (method.isAnnotationPresent(GET.class)) {
-                info.method = "GET";
-                info.relativePath = method.getAnnotation(GET.class).value();
+                methodProxyDetails.setMethodAndRelativePath("GET", method.getAnnotation(GET.class).value());
```

Comment:
I usually like to keep all of my parsing logic to a single file so that it's easier to keep it all in my head. However, I like the idea of lazily parsing a swagger interfaces since I'm sure that these interfaces can get pretty large and that may produce a performance impact. Let me make this change in a future pull request.